purchase_analytic_distribution/views/purchase_view.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+<odoo>
+
+    <record model="ir.ui.view" id="purchase_order_line_form_inherit">
+        <field name="name">purchase.order.line.form.inherit</field>
+        <field name="model">purchase.order.line</field>
+        <field name="inherit_id" ref="purchase.purchase_order_line_form2"/>
+        <field name="arch" type="xml">
+            <field name="account_analytic_id" position="before">
+                <field name="analytic_distribution_id"  groups="purchase.group_analytic_accounting"/>
+            </field>
+        </field>
+    </record>
+    <record model="ir.ui.view" id="purchase_order_form_analytic_inherit">
+        <field name="name">purchase.order.form.analytic.inherit</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/tree//field[@name='account_analytic_id']" position="before">
+                <field name="analytic_distribution_id" groups="purchase.group_analytic_accounting"/>
+            </xpath>
+            <xpath expr="//field[@name='order_line']/form//field[@name='account_analytic_id']" position="before">
+                <field name="analytic_distribution_id" groups="purchase.group_analytic_accounting"/>
+            </xpath>
+        </field>
+    </record>
+    <record id="view_purchase_order_form_editable_list" model="ir.ui.view">
+        <field name="name">purchase.order.form.editable.list</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form"/>
+        <field name="groups_id" eval="[(6, 0, [ref('product.group_uom'), ref('purchase.group_analytic_accounting')])]"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/tree" position="attributes">
+                <attribute name="editable"/>
```

Comment:
Why removing the editable attribute? You can put the analytic distribution in the tree itself.